internal/controller/driver_controller.go

```diff
@@ -17,46 +17,1211 @@ limitations under the License.
+								},
+								Resources: ptr.Deref(
+									pluginSpec.Resources.Addons,
+									corev1.ResourceRequirements{},
+								),
+							})
+							// Liveness Sidecar Container
+							if r.driver.Spec.Liveness != nil {
+								containers = append(containers, corev1.Container{
+									Name:            "liveness-prometheus",
+									Image:           r.images["plugin"],
+									ImagePullPolicy: imagePullPolicy,
+									SecurityContext: &corev1.SecurityContext{
+										Privileged: ptr.To(true),
+										Capabilities: &corev1.Capabilities{
+											Drop: []corev1.Capability{"All"},
+										},
+									},
+									Args: []string{
+										utils.TypeContainerArg("liveness"),
+										utils.EndpointContainerArg,
+										utils.MetricsPortContainerArg(r.driver.Spec.Liveness.MetricsPort),
+										utils.MetricsPathContainerArg,
+										utils.PoolTimeContainerArg,
+										utils.TimeoutContainerArg(3),
+									},
+									Env: []corev1.EnvVar{
+										utils.PodIpEnvVar,
+									},
+									VolumeMounts: []corev1.VolumeMount{
+										utils.PluginDirVolumeMount,
+									},
+									Resources: ptr.Deref(
+										pluginSpec.Resources.Liveness,
+										corev1.ResourceRequirements{},
+									),
+								})
+							}
+						}
+						return containers
+					}),
+					Volumes: utils.Call(func() []corev1.Volume {
+						volumes := append(
+							// Add user defined volumes at the start to make sure they do not
+							// overwrite built in volumes.
+							utils.MapSlice(
+								pluginSpec.Volumes,
+								func(v csiv1a1.VolumeSpec) corev1.Volume {
+									return v.Volume
+								},
+							),
+							utils.HostDevVolume,
+							utils.HostSysVolume,
+							utils.HostRunMountVolume,
+							utils.LibModulesVolume,
+							utils.KeysTmpDirVolume,
+							utils.PluginDirVolume(pluginSpec.KubeletDirPath, r.driverPrefix),
+							utils.PluginMountDirVolume(pluginSpec.KubeletDirPath),
+							utils.PodsMountDirVolume(pluginSpec.KubeletDirPath),
+							utils.RegistrationDirVolume(pluginSpec.KubeletDirPath),
+						)
+						if ptr.Deref(pluginSpec.EnableSeLinuxHostMount, false) {
+							volumes = append(
+								volumes,
+								utils.EtcSelinuxVolume,
+							)
+						}
+						if r.driver.Spec.Encryption != nil {
+							volumes = append(
+								volumes,
+								utils.KmsConfigVolume(&r.driver.Spec.Encryption.ConfigMapRef),
+							)
+						}
+						if r.isRdbDriver() {
+							volumes = append(
+								volumes,
+								utils.OidcTokenVolume,
+							)
+						}
+						return volumes
+					}),
+				},
+			},
+		}
+
+		return nil
+	})
+
+	logCreateOrUpdateResult(log, "node plugin daemonset", daemonSet, opResult, err)
+	return err
+}
+
+func (r *driverReconcile) reconcileLivnessService() error {
+	if r.driver.Spec.Liveness == nil {
+		return nil
+	}
+
+	service := &corev1.Service{}
+	service.Namespace = r.driver.Namespace
+	service.Name = r.generateName("livness")
```

Comment:
`livness` to `liveness` need to be changed in all the places

Reply:
ack

Reply:
done